cluster-management/resource-management/cluster_resource_report.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+import os, json
+import requests
+from tempfile import NamedTemporaryFile
+import socket
+ 
+# Primary Function: This script is designed to log into an OpenShift cluster, read the kube config
+# and then retrieve the resources that each pod requests. It produces summed output based on each
+# host
+# Date: Nov 8, 2017
+ 
+ 
+def query_cluster_api(master_fqdn):
+    """
+    :param master_fqdn: the only parameter is the master fqdn, or the openshift master public hostname
+    :return: returns a list of "strings" which are actually json objects to be processed by json.loads()
+    """
+    response_list = []
+    # The certificate section below could be done with the python requests module, but since we are
+    # exceuting shell commands through a remote host, it is set up to use the 'curl' command
+    kube_config_location = "/root/.kube/config"
+    client_cert_list_compre = '[word.split(": ")[1].strip() for word in open("%s").readlines() ' \
+                              'if "client-certificate-data" in word]' % kube_config_location
+    client_key_list_compre = '[word.split(": ")[1].strip() for word in open("%s").readlines() ' \
+                             'if "client-key-data" in word]' % kube_config_location
+ 
+    client_cert_base64 = os.popen("sudo python -c 'print %s'" % client_cert_list_compre).read()
+    client_cert_decoded = client_cert_base64.strip().strip("[").strip("]").strip("'").decode('base64')
+ 
+    client_key_base64 = os.popen("sudo python -c 'print %s'" % client_key_list_compre).read()
+    client_key_decoded = client_key_base64.strip().strip("[").strip("]").strip("'").decode('base64')
+ 
+    pod_url = "https://%s:8443/api/v1/namespaces/default/pods" % master_fqdn
```

Comment:
Did you want to query all pods in the cluster instead of in the `default` project?

If so, the `https://%s:8443/api/v1/pods` can be used

Reply:
in this case i did not because client was using default for the pods however this can be adjusted